Feature/sensor calibration tool chain (#4)
pixmoving-moveit/pixkit-documentation-en: docs/sensor-calibration/LiDAR-camera-calibration.md

## Objectives of this chapter

- Understand extrinsic calibration
- Master the calibration skills of LiDAR to camera
- Calibrate qualified extrinsic parameters

## Calibration of the vehicle
- hook 
    - Top LiDAR [RS-Helios-16P]
    - Front camera [SG2-OX03CC-5200-GMSL2F-H120]

## Prerequisites

- Completed [camera intrinsic calibration](./camera-intrisics-calibration.md)
- Find a suitable calibration scene
    - There is a sedan in the camera image
    - There is a utility pole in the camera image
    - There is a wall with windows in the camera image

![](./image/lidar2camera/cali_result.jpg)

## Start calibration

> ### step-1: Collect calibration data

Reviewer: 加入标定工具安装文档的链接

Reviewer: 没跟安装标定工具的文档联系起来

Author: 已完成

| config/top\_center\_lidar-to-center\_camera-extrinsic.json | Initial extrinsic file | Default value can be used |
| config/parser.py | Script to generate sensors\_calibration.yaml file | Generates parameter file usable by Autoware |
| config/sensors\_calibration.yaml | Extrinsic parameter file | Parameter file usable by Autoware |
| config/example.jpg | Example result of calibration output | Automatically filled in after completing [camera intrinsic calibration](./camera-intrisics-calibration.md) |
| config/example.txt | Example result of calibration output | Automatically filled in after completing [camera intrinsic calibration](./camera-intrisics-calibration.md) |

#### calibration_script/lidar2camera/lidar2camera.yaml 

![](./image/lidar2camera/configuration_file.jpg)

| Parameter Name | Parameter Function |
| --- | --- |
| pcd\_path | The input path of point cloud data required by the calibration program. |
| png\_path | The input path of images required by the calibration program. |

### Start calibration

```shell
./calibration_script/lidar2camera/lidar2camera.sh
```
![](./image/lidar2camera/lidar2camra1.gif)
![](./image/lidar2camera/lidar2camera2.gif)

> ### step-3: Parameter configuration

Reviewer: ```suggestion
### step-3: Intrinsics/Extrinsics configuration
```

![](./image/lidar2camera/cali_result.jpg)

You can see that in the picture, the utility pole, the vehicle plate and the point cloud of the window are all aligned. This is the result of calibration. If you want to achieve such a result, you need to adjust the parameters to achieve such a calibration effect. You need to click the program button to fine-tune it.

> #### External Parameter Adjustment

Reviewer: ```suggestion
#### Extrinsics Adjustment
```

| Button | Description |
| --- | --- |
| +x degree | Adjust the roll angle |
| +y degree | Adjust the pitch angle |
| +z degree | Adjust the yaw angle |
| +x trans | Adjust the x-axis translation |
| +y trans | Adjust the y-axis translation |
| +z trans | Adjust the z-axis translation |

#### Camera Intrinsic Adjustment

| Button | Description |
| --- | --- |
| \+ fy | The camera intrinsic can be left at default |
| \+ fx | The camera intrinsic can be left at default |

<kbd>Intensity Color</kbd>: LiDAR intensity is a parameter that records the intensity of the laser beam's return, which allows us to see the intensity of the point cloud under different materials.

<kbd>Overlap Filter</kbd>: Eliminates overlapping LiDAR points within 0.4m in depth.

<kbd>deg step</kbd> <kbd>t step</kbd> <kbd>fxfy scale</kbd>: These three buttons can adjust the step size of each click or keyboard input - default is recommended.

<kbd>point size</kbd>: Adjusts the size of LiDAR points in the projected image.

<kbd>Reset</kbd>: Click this button to reset all manual adjustments and restore the initial parameters.

> <kbd>Save Image</kbd>: When the calibration is completed, click this button, and the calibration image, external and internal parameter matrix will be stored in `calibration_script/lidar2camera/config` by default.

Reviewer: ```suggestion
<kbd>Save Image</kbd>: When the calibration is completed, click this button, and the calibration image, extrinsics and intrinsics matrix will be stored in `calibration_script/lidar2camera/config` by default.
```
